app/src/test/java/ch/epfl/sdp/TimedItemTest.java

```diff
@@ -0,0 +1,104 @@
+package ch.epfl.sdp;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import ch.epfl.sdp.entity.Player;
+import ch.epfl.sdp.game.Game;
+import ch.epfl.sdp.item.Scan;
+import ch.epfl.sdp.item.Shield;
+import ch.epfl.sdp.item.Shrinker;
+import ch.epfl.sdp.item.TimedItem;
+import ch.epfl.sdp.map.GeoPoint;
+import ch.epfl.sdp.map.MapsActivity;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class TimedItemTest {
+    public final int FPS = 30;
+    public final int countTime = 2;
+    Player player;
+
+    @Before
+    public void setup(){
+        player = new Player();
+    }
+
+    @Test
+    public void timedItemCounterIsDecrementedBy1WhenUpdated(){
+        TimedItem timedItem = new TimedItem(null, null, false, null, countTime) {};
+
+        for(int i = countTime*FPS; i > 0; --i) {
+            assertEquals(timedItem.getRemainingTime(), i/FPS);
+            timedItem.update();
+            assertEquals(timedItem.getRemainingTime(), (i-1)/FPS);
+        }
+    }
+
+    @Test
+    public void timedItemIsDeletedFromTheListWhenTimeIsFinished(){
+        TimedItem timedItem = new TimedItem(null, null, false, null, countTime) {};
+        Game game = new Game(null);
+        Game.addToUpdateList(timedItem);
+
+        for(int i = countTime*FPS; i > 0; --i) {
+            timedItem.update();
+        }
+
+        timedItem.update();
+    }
+
+    @Test
+    public void scanGetsUpdated(){
+        Scan scan = new Scan(null, false, countTime);
+
+        scan.update();
+        // Cannot assert that mapApi has changed...
```

Comment:
Use the MockMapApi class in the test folder. We can try to emulate the update.

Reply:
For that I would need a method to set the MapApi in MapsActivity.